src/main/groovy/nebula/plugin/info/reporting/InfoJarPropertiesFilePlugin.groovy

```diff
@@ -34,35 +35,36 @@ class InfoJarPropertiesFilePlugin implements Plugin<Project>, InfoReporterPlugin
-                    //temp.properties is renamed later when placed into jar
-                    def taskName = jarTask.name.capitalize()
-                    File propertiesFile = new File(project.buildDir, "properties_for_${taskName}/temp.properties")
-                    Task prepareFile = project.tasks.create("createPropertiesFileFor${taskName}") { Task task ->
-                        task.outputs.file(propertiesFile)
-                        task.doLast {
-                            //Task action is intentionally creating empty file.
-                            propertiesFile.text = ""
-                        }
-                    }
+            //we cannot use the right module name because touching manifest task to early causes incorrect name computation
+            //temp.properties is renamed later when placed into jar
+            File propertiesFile = new File(project.buildDir, "properties_for_jar/temp.properties")
+            TaskProvider<Task> prepareFile = project.tasks.register("createPropertiesFileForJar") { Task task ->
+                task.outputs.file(propertiesFile)
+                task.doLast {
+                    //Task action is intentionally creating empty file.
+                    propertiesFile.text = ""
+                }
+            }
+
+            project.tasks.withType(Jar).configureEach { Jar jarTask ->
+                //explicit dependency on original task to keep contract that `jar` task invocation will produce properties file
+                //even this file is actually not bundled into jar
+                jarTask.dependsOn(manifestTask)
 
-                    //we link the output file from the task to the spec to add it into jar, but the file is empty, it will
-                    //help to ignore changes in its content for caching
-                    jarTask.metaInf { CopySpec spec ->
-                        spec.from(prepareFile.outputs).rename("temp.properties", manifestTask.propertiesFile.name)
-                    }
-                    jarTask.doFirst {
-                        //when we are after all caching decisions we fill the file with all the data
-                        new PropertiesWriter().writeProperties(propertiesFile, project)
-                    }
+                //we link the output file from the task to the spec to add it into jar, but the file is empty, it will
+                //help to ignore changes in its content for caching
+                jarTask.metaInf { CopySpec spec ->
+                    spec.from(prepareFile).rename("temp.properties", manifestTask.get().propertiesFile.name)
+                }
+                jarTask.doFirst {
+                    //when we are after all caching decisions we fill the file with all the data
+                    new PropertiesWriter().writeProperties(propertiesFile, project)
+                }
+                jarTask.doLast {
```

Comment:
This is the ugliest part :( previously each Jar task had its own properties file. To be able to make it lazy I had to switch back to a single `prepareFile` task per module. However, to keep using caching trick each jar task has to clean the file so the following jar task can use the same caching trick. The caching trick is that cache sees only empty file all the time so it is not invalidated by changing properties.